src/exif/service.rs

```diff
@@ -0,0 +1,172 @@
+                return None;
+            }
+        };
+        let exif: Exif = iter.into();
+        exif.get(ExifTag::DateTimeOriginal)
+            .map(|it| it.as_time())
+            .flatten()
+            .map(|it| it.naive_local())
+    } else if ms.has_track() {
+        let data: TrackInfo = match parser.parse(ms) {
+            Ok(td) => td,
+            Err(_) => return None,
+        };
+        data.get(nom_exif::TrackInfoTag::CreateDate)
+            .map(|it| it.as_time())
+            .flatten()
+            .map(|it| it.naive_local())
+    } else {
+        None
+    }
+}
+
+/// Processes a single file to extract EXIF data and update its creation date in the database.
+///
+/// ## Parameters
+/// * `file_id` - The ID of the file to process
+///
+/// ## Returns
+/// * `true` if the message should be removed from the queue (success or unrecoverable error)
+/// * `false` if the message should be re-queued (temporary failure)
+pub async fn process_single_file_exif(message_data: String) -> bool {
+    let id: u32 = match message_data.parse() {
+        Ok(i) => i,
+        Err(e) => {
+            log::error!(
+                "Failed to parse {message_data} as a u32! Exception is {e:?}\n{}",
+                Backtrace::force_capture()
+            );
+            // Invalid message, don't re-queue
+            return true;
+        }
+    };
+
+    // Get file metadata from database
+    let con = open_connection();
+    let file_record = match file_repository::get_file(id, &con) {
+        Ok(f) => f,
+        Err(e) => {
+            log::debug!(
+                "Failed to get file from database when processing EXIF data. File id [{id}]; error is {e:?}",
+            );
+            con.close().unwrap();
+            // File doesn't exist in DB, don't requeue
+            return true;
+        }
+    };
+
+    // Only process image and video files
+    match file_record.file_type {
+        FileTypes::Image | FileTypes::Video => {}
+        _ => {
+            con.close().unwrap();
+            // Not an image/video, no EXIF to process
+            return true;
+        }
+    }
+
+    // Get file path on disk
+    let path = match get_file_path(id) {
+        Ok(p) => {
+            let path = PathBuf::from(file_dir()).join(p);
+            path.to_string_lossy().into_owned()
+        }
+        Err(_) => {
+            log::debug!("File id {id} not found on disk, skipping EXIF processing");
+            con.close().unwrap();
+            // File is no longer on disk, don't requeue
+            return true;
+        }
+    };
+
+    // Parse EXIF date or use current date as fallback
+    let create_date =
+        parse_exif_date(&path).unwrap_or_else(|| chrono::offset::Local::now().naive_local());
+
+    // Update file record in database with the extracted date
+    let update_result = con.execute(
+        "UPDATE FileRecords SET dateCreated = ?1 WHERE id = ?2",
+        rusqlite::params![create_date, id],
+    );
```

Comment:
add a `repository.rs` file that contains all exif writes to the database. Include necessary tests